### ISSUE TITLE
Extensions specified in devcontainer.json not being installed

### ISSUE THREAD

**Issue by rjra100** [microsoft/vscode-remote-release#4839]
Issue Type: <b>Bug</b>

After upgrading to VSCode 1.55, most extensions we specify in our devcontainer.json are no longer being installed.

In the log, we have a call to `server.sh --force-disable-user-env --use-host-proxy --disable-telemetry --port 0 --extensions-download-dir <path to extensionsCache> --install-extension <blah> --install-extension <blah2> ... --start-server`, but the only extension that actually seems to be installed is one installed directly from a .vsix built into our Docker image - nothing from the marketplace appears to show up as even attempting to install. Other than that one (a custom build of vadimcn.vscode-lldb), no extensions show up in the extension pane as installed in the container.

Although I don't see anything similar in the log, manually running the `server.sh --install-extension` command in a VSCode terminal gives:

```
[15:09:57] Extension host agent started.
Installing extensions...
[15:10:18] Error: connect ECONNREFUSED 13.107.42.18:443
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16) {
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '13.107.42.18',
  port: 443
}
```

Removing the `--force-disable-user-env` switch appears to successfully download the extensions (though they don't work when run from the terminal like this; I'm guessing they're loaded into the wrong extension host), so it's possibly some sort of problem with our proxy setup. However, everything worked fine in VSCode 1.54.3.

The problem may well lie in VSCode itself, but I don't know enough about the interaction between Remote-Containers and VSCode to put together a coherent bug report there, so I'd appreciate some help :)

Extension version: 0.163.2
VS Code version: Code 1.55.0 (c185983a683d14c396952dd432459097bc7f757f, 2021-03-30T16:01:55.261Z)
OS version: Windows_NT x64 10.0.19041
Remote OS version: Linux x64 4.19.128-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.86GB (2.74GB free)|
|Process Argv|--crash-reporter-id 3b02320b-d5a4-4fea-91a2-32fa216b37ff|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container: portal_quantirvols-dev|
|OS|Linux x64 4.19.128-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|Memory (System)|9.73GB (2.99GB free)|
|VM|0%|
</details>

I'm attaching logs from both VSCode 1.54.3 (worked fine) and VSCode 1.55.0 (most extensions missing)

[extensions_vscode_1.54.3.txt](https://github.com/microsoft/vscode-remote-release/files/6265714/extensions_vscode_1.54.3.txt)
[extensions_vscode_1.55.txt](https://github.com/microsoft/vscode-remote-release/files/6265715/extensions_vscode_1.55.txt)

**Comment by rjra100** [comment]
May be a dup of https://github.com/microsoft/vscode-remote-release/issues/4812 - sorry, hadn't found that one.

**Comment by chrmarti** [comment]
This should be fixed in VS Code Insiders (https://code.visualstudio.com/insiders/) and the fix will be in the upcoming 1.55.1 release of VS Code. (Tracked as https://github.com/microsoft/vscode/issues/120332)
